language/move-vm/types/src/natives/function.rs

```diff
@@ -56,6 +56,8 @@ pub trait NativeContext {
     fn save_event(&mut self, event: ContractEvent) -> VMResult<()>;
     /// Converts types to fet types.
     fn convert_to_fat_types(&self, types: Vec<Type>) -> VMResult<Vec<FatType>>;
+    /// Whether a type is a resource of not.
+    fn is_resource(&self, ty: &Type) -> VMResult<bool>;
```

Comment:
I would expect this type to be fully instantiatied always and no need to go through the resolver ever.
I am fine fixing it later but I am wondering if I am missing something or this was mostly to have the API be the most general possible (which may or may not be a good thing)

Reply:
I added this because it was the most straightforward way for me to determine the kind of a generic type, but let me check if I can instead make the `Type` carry its kind so we don't have to go through the loader again.